src/client/Client.cc

```diff
@@ -10716,7 +10716,7 @@ void Client::C_Read_Finisher::finish_io(int r)
 {
   utime_t lat;
 
-  // Caller holds client_lock so we don't need to take it.
```

Comment:
so you mean to say there are callers which dont hold the client lock?

Reply:
Yeah! In nfsganesha usecase, it's observed that the call back are called without client_lock being held leading to corruption.

Reply:
so what happens with a code path such as `Client::ll_preadv_pwritev` (async i/o) which holds client lock (`std::unique_lock cl(client_lock);`) ? Is it fine that the multiple code paths hold the client lock (async call and call back in this case)?

Reply:
As long it's not deadlock, we should be good.

Reply:
@dparmar18 ptal at https://github.com/ceph/ceph/pull/59987#issuecomment-2378781787